sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -421,6 +422,119 @@ ng_ipv6_addr_t *ng_ipv6_netif_find_best_src_addr(kernel_pid_t pid, const ng_ipv6
     return _match_prefix(pid, dest, true);
 }
 
+/* TODO: put this somewhere more central and L2 protocol dependent */
+#define IID_LEN (8)
+
+static bool _hwaddr_to_iid(uint8_t *iid, const uint8_t *hwaddr, size_t hwaddr_len)
+{
+    uint8_t i = 0;
+
+    memset(iid, 0, IID_LEN);
+
+    switch (hwaddr_len) {
+        case 4:
+            iid[0] = hwaddr[i++];
+            iid[0] ^= 0x02;
+            iid[1] = hwaddr[i++];
+
+        case 2:
+            iid[6] = hwaddr[i++];
+
+        case 1:
+            iid[3] = 0xff;
+            iid[4] = 0xfe;
+            iid[7] = hwaddr[i++];
+            break;
+
+        case 6:
+            iid[0] = hwaddr[i++];
+            iid[0] ^= 0x02;
+            iid[1] = hwaddr[i++];
+            iid[2] = hwaddr[i++];
+            iid[3] = 0xff;
+            iid[4] = 0xfe;
+            iid[5] = hwaddr[i++];
+            iid[6] = hwaddr[i++];
+            iid[7] = hwaddr[i++];
+            break;
+
+        case 8:
+            iid[0] = hwaddr[i++];
+            iid[0] ^= 0x02;
+            iid[1] = hwaddr[i++];
+            iid[2] = hwaddr[i++];
+            iid[3] = hwaddr[i++];
+            iid[4] = hwaddr[i++];
+            iid[5] = hwaddr[i++];
+            iid[6] = hwaddr[i++];
+            iid[7] = hwaddr[i++];
+            break;
+
+        default:
+            DEBUG("Unknown hardware address length\n");
+            return false;
+    }
+
+    return true;
+}
+
+void ng_ipv6_netif_init_by_dev(void)
+{
+    kernel_pid_t *ifs;
+    size_t ifnum;
+
+    ifs = ng_netif_get(&ifnum);
+
+    for (size_t i = 0; i < ifnum; i++) {
+        ng_ipv6_addr_t addr;
+        uint16_t hwaddr_len = 0;
+        uint8_t hwaddr[8];
+        bool try_long = false;
+        ng_ipv6_netif_t *ipv6_if = ng_ipv6_netif_get(ifs[i]);
+        int res = 0;
+
+        if (ipv6_if == NULL) {
+            continue;
+        }
+
+        mutex_lock(&ipv6_if->mutex);
+
+#ifdef MODULE_NG_SIXLOWPAN
+        ng_nettype_t if_type = NG_NETTYPE_UNDEF;
+
+        if ((ng_netapi_get(ifs[i], NETCONF_OPT_PROTO, 0, &if_type,
+                           sizeof(if_type)) != -ENOTSUP) &&
+            (if_type == NG_NETTYPE_SIXLOWPAN)) {
```

Comment:
... As e. g. Ethertype. I think however the doc still needs updating in this regard.